File: group_anagrams.js
const groupAnagrams = function(strings) {
    //write your code here
    const sorted = strings.map(x => x.split('').sort().join(''));
    const ht = {};

    for (let i = 0; i < strings.length; i++) {
        if (!ht[sorted[i]]) ht[sorted[i]] = [strings[i]];
        else ht[sorted[i]].push(strings[i]);

    }
    return Object.values(ht);
}

File: group_anagrams.py
def group_anagrams(strings):
    #write your code here
    if len(strings)==0:
        return []
        
    sorted_string = [''.join(sorted(i)) for i in strings]
    
    hash={}
    
    for i in range(len(sorted_string)):
        if(sorted_string[i] in hash):
            hash[sorted_string[i]].append(strings[i])
        else:
            hash[sorted_string[i]]=[strings[i]]
        
    return list(hash.values())